controllers/memberoperatorconfig/memberoperatorconfig_controller.go

```diff
@@ -105,9 +104,24 @@ func (r *Reconciler) handleWebhookDeploy(ctx context.Context, cfg membercfg.Conf
 		if err := deploy.Webhook(ctx, r.Client, r.Client.Scheme(), namespace, webhookImage); err != nil {
 			return err
 		}
+		// TODO --  temporary migration step to delete the webhook with the old names
+		deleted, err := deploy.Delete(ctx, r.Client, r.Client.Scheme(), namespace, true)
+		if err != nil {
+			return err
+		}
+		if deleted {
+			logger.Info("Deleted previously deployed webhook app")
+		}
+		// TODO -- end temporary migration
 		logger.Info("(Re)Deployed users' pods webhook")
 	} else {
-		logger.Info("Skipping deployment of users' pods webhook")
+		deleted, err := deploy.Delete(ctx, r.Client, r.Client.Scheme(), namespace, false)
+		if err != nil {
+			return err
+		}
+		if deleted {
+			logger.Info("Deleted previously deployed webhook app")
+		}
```

Comment:
Yes, because we now deploy the member operator before we disable the configuration see: https://github.com/codeready-toolchain/toolchain-e2e/pull/950/files#diff-163703aeb25751c1605d305fbfdf39f3a9dfb74d2ac9d8a03f5cf0169d7d41b7R292 . 

See also my comment here : https://github.com/codeready-toolchain/toolchain-e2e/pull/950/files#r1580269009

Reply:
Thanks for the explanation, I see your point now. I think the webhook provides some pretty important functionality that I'm not sure we'd want to delete unless it's explicitly meant to be deleted but I can't think of a reason why we'd disable the configuration and expect the webhook to stay as it was so I guess it's fine. I'd just suggest we update [the comment in the api](https://github.com/codeready-toolchain/api/blob/49462fdf229a6a392cae4bd4d4f4b551e8870f35/api/v1alpha1/memberoperatorconfig_types.go#L158) to make it clear what happens.

Reply:
> I'd just suggest we update [the comment in the api](https://github.com/codeready-toolchain/api/blob/49462fdf229a6a392cae4bd4d4f4b551e8870f35/api/v1alpha1/memberoperatorconfig_types.go#L158) to make it clear what happens.

good point, please see https://github.com/codeready-toolchain/api/pull/425/files

Reply:
> I think the webhook provides some pretty important functionality that I'm not sure we'd want to delete unless it's explicitly meant to be deleted but I can't think of a reason why we'd disable the configuration and expect the webhook to stay

I was discussing this with @MatousJobanek, a better approach IMO would be to decouple the deployment of the webhooks from member operator , this would allow to have more flexibility and would avoid such "workarounds" where we deploy something and then we need to delete it. But the decoupling of the webhooks is a bigger effort and not on the priority list atm.